torchao/prototype/mx_formats/mx_subclass.py

```diff
@@ -151,7 +156,116 @@ def _mx_inference_linear_transform(
     return module
 
 
+def _get_nvfp4_dtype():
+    """Factory function for NVFP4 dtype defaults."""
+    if not TORCH_VERSION_AT_LEAST_2_8:
+        raise RuntimeError("NVFP4InferenceConfig requires PyTorch 2.8 or later")
+    return torch.float4_e2m1fn_x2
+
+
+@dataclass
+class NVFP4InferenceConfig(AOBaseConfig):
+    """
+    NVIDIA FP4 (NVFP4) Inference Quantization Configuration
+
+    This is a specialized configuration for NVIDIA's FP4 format with UE4M3 scales.
+    It provides defaults optimized for NVFP4:
+    - Data: float4_e2m1fn_x2
+    - Scales: float8_e4m3fn (UE4M3)
+    - Block size: 16 (required for NVFP4)
+    - CUBLAS kernel (optimized for VEC16_UE4M3)
+    """
+
+    block_size: int = 16  # NVFP4 requires block size 16
+
+    # NVFP4 uses FP4 data
+    activation_dtype: torch.dtype = field(default_factory=_get_nvfp4_dtype)
+    weight_dtype: torch.dtype = field(default_factory=_get_nvfp4_dtype)
+
+    # NVFP4 uses E4M3 scales
+    scale_dtype: torch.dtype = torch.float8_e4m3fn
+
+    # CUBLAS is preferred for NVFP4 with VEC16_UE4M3 support
+    gemm_kernel_choice: MXGemmKernelChoice = MXGemmKernelChoice.CUBLAS
+
+    # Set some magic perf settings
+    set_inductor_config: bool = False
```

Comment:
nit: move outside of this API? It's convenient to have it here but also breaks separation of concerns.